.github/workflows/ci-kuberenetes-itests.yml

```diff
@@ -0,0 +1,97 @@
+          restore-keys: ${{ runner.os }}-maven-
+      - name: Install artifacts
+        run: mvn -f pom.xml ${MAVEN_ARGS} -DskipTests clean install -pl :quarkus-integration-test-kubernetes-invoker -am
+  kubernetes:
+    name: Kubernetes Integration Tests
+    needs: cache
+    if: github.event.label.name == 'area/kubernetes' || contains(github.event.pull_request.labels.*.name, 'area/kubernetes')
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes: [v1.20.1]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Setup Minikube-Kubernetes
+        uses: manusa/actions-setup-minikube@v2.4.1
+        with:
+          minikube version: v1.16.0
+          kubernetes version: ${{ matrix.kubernetes }}
+          github token: ${{ secrets.GITHUB_TOKEN }}
+          start args: '--addons=metrics-server --force'
+      - name: Cache .m2 registry
+        uses: actions/cache@v2.1.6
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-maven-
+      - name: Setup Java 11 
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Run Kubernetes Invoker Tests
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          export QUARKUS_CONTAINER_IMAGE_GROUP=${{ github.actor }}
+          export QUARKUS_CONTAINER_IMAGE_REGISTRY=ghcr.io
+          mvn -B clean install -pl :quarkus-integration-test-kubernetes-invoker -Pkubernetes-e2e-tests
+
+  openshift:
+    name: Openshift Integration Tests
+    needs: cache
+    if: github.event.label.name == 'area/kubernetes' || contains(github.event.pull_request.labels.*.name, 'area/kubernetes')
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        openshift: [v3.11.0]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Setup OpenShift
+        uses: manusa/actions-setup-openshift@v1.1.3
+        with:
+          oc version: ${{ matrix.openshift }}
+          github token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Cache .m2 registry
+        uses: actions/cache@v2.1.5
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-maven-
+      - name: Setup Java 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Run Openshift Invoker Tests
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          export QUARKUS_CONTAINER_IMAGE_GROUP=${{ github.actor }}
```

Comment:
wont you need a way to make the published image unique - i.e. based on which pr/branch its running againts to avoid collisions?

Reply:
That's a very good point. Maybe, we should set the version to something that includes the pr number, timestamp etc.

Reply:
let's try with the sha